Sources/SpeziOpenAI/MessageInputView.swift

```diff
@@ -54,6 +57,37 @@ public struct MessageInputView: View {
                 .padding(.trailing, -38)
                 .padding(.bottom, 3)
                 .disabled(message.isEmpty)
+            if speechRecognizer.isAvailable {
+                Button(
+                    action: {
+                        microphoneButtonPressed()
+                    }
+                ) {
+                    ZStack {
+                        Circle()
+                            .foregroundColor(speechRecognizer.isRecording ? Color.red : Color.blue)
+                            .frame(width: 44, height: 44)
+                        Image(systemName: "mic.fill")
+                            .foregroundColor(.white)
+                            .font(.title)
+                            .frame(width: 44, height: 44)
+                            .background(Color.clear)
+                            .alignmentGuide(HorizontalAlignment.center, computeValue: { dimension in
+                                dimension[HorizontalAlignment.center]
+                            })
+                            .alignmentGuide(VerticalAlignment.center, computeValue: { dimension in
+                                dimension[VerticalAlignment.center]
+                            })
+                            .scaleEffect(speechRecognizer.isRecording ? 1.2 : 1.0)
+                            .opacity(speechRecognizer.isRecording ? 0.7 : 1.0)
+                            .animation(
+                                speechRecognizer.isRecording ? .easeInOut(duration: 0.5).repeatForever(autoreverses: true) : .default,
+                                value: speechRecognizer.isRecording
+                            )
+                    }
+                }
+                .padding(EdgeInsets(top: 5, leading: 20, bottom: 0, trailing: 0))
```

Comment:
I like the usage of the microphone image as a good representation of the usage of the dictation functionality. As discussed in our meeting on Thursday, here is some UI feedback. 
In its current form, the dictation button is bigger than the text field and the send button. I would suggest that we keep it at the same size as the send button to make this consistent with the rest of the UI in the input field.

One even cooler approach would be to replicate the behaviour of the iOS Messages app: If there is not text entered in the field, the send button turns into a microphone button that starts of dictation:
![IMG_0429](https://github.com/StanfordSpezi/SpeziML/assets/28656495/be8f1e5c-e56d-44e5-ba2a-6dbf71103785)

I could imagine the same behaviour here: We build an if/else statement around the use the same UI as the end button for the microphone button. If there is no text in the field, we display the microphone button, otherwise the send button. Reusing the views (or using the same underlying view) would then ensure that they also have the same size. 

The background of the microphone button could be based on the state:
- Not enabled: No background as with the iMessage textfield
- If dilation is running, we add the nice red background that you currently have and the let the microphone pulse as we currently have it doing to show the nice diction effect.

This approach also simplifies the use cases where a user could press send while dictating as possible in the current UI. if the send button is not there, we have full control to properly stop the dictation session before sending.

What do you think?

Reply:
@AdritRao I took a look at the changes and due to the time pressure to get this merged I just went ahead and implemented these changes myself.